extensions/map-layers/src/ui/widget/MapLayerSettingsMenu.tsx

```diff
@@ -0,0 +1,93 @@
+      if (undefined !== indexInDisplayStyle) {
+        hasRange = (undefined !== await activeViewport.displayStyle.getMapLayerRange(indexInDisplayStyle, mapLayerSettings.isOverlay));
+      }
+      setHasRangeData(hasRange);
+    }
+    fetchRangeData(); // eslint-disable-line @typescript-eslint/no-floating-promises
+  }, [activeViewport, mapLayerSettings]);
+
+  const onSettingsClick = React.useCallback(() => {
+    setIsSettingsOpen((prev) => !prev);
+  }, [setIsSettingsOpen]);
+
+  const handleCloseSetting = React.useCallback(() => {
+    setIsSettingsOpen(false);
+  }, [setIsSettingsOpen]);
+
+  const handleRemoveLayer = React.useCallback(() => {
+    setIsSettingsOpen(false);
+    onMenuItemSelection("delete", mapLayerSettings);
+  }, [setIsSettingsOpen, onMenuItemSelection, mapLayerSettings]);
+
+  const handleZoomToLayer = React.useCallback(() => {
+    setIsSettingsOpen(false);
+    onMenuItemSelection("zoom-to-layer", mapLayerSettings);
+  }, [setIsSettingsOpen, onMenuItemSelection, mapLayerSettings]);
+
+  const applyTransparencyChange = React.useCallback((value: number) => {
+    if (activeViewport) {
+      const newTransparency = value;
+      const displayStyle = activeViewport.displayStyle;
+      const indexInDisplayStyle = displayStyle.findMapLayerIndexByNameAndUrl(mapLayerSettings.name, mapLayerSettings.url, mapLayerSettings.isOverlay);
+      if (-1 !== indexInDisplayStyle) {
+        const styleTransparency = displayStyle.mapLayerAtIndex(indexInDisplayStyle, mapLayerSettings.isOverlay)?.transparency;
+        const styleTransparencyValue = styleTransparency ? styleTransparency : 0;
+        if (Math.abs(styleTransparencyValue - newTransparency) > 0.01) {
+          // update the display style
+          displayStyle.changeMapLayerProps({ transparency: newTransparency }, indexInDisplayStyle, mapLayerSettings.isOverlay);
+          activeViewport.invalidateRenderPlan();
+
+          // force UI to update
+          // loadMapLayerSettingsFromStyle(activeViewport.displayStyle);
+        }
+      }
+    }
+  }, [activeViewport, mapLayerSettings]);
+
+  const handleTransparencyChange = React.useCallback((values: readonly number[]) => {
+    if (values.length) {
+      const newTransparency = values[0] / 100.0;
+      if (newTransparency !== transparency) {
+        setTransparency(newTransparency);
+        applyTransparencyChange(newTransparency);
+      }
+    }
+  }, [transparency, applyTransparencyChange]);
+
+  return (
+    <>
+      <button data-testid="map-layer-settings" className="map-layer-settings icon icon-more-vertical-2" ref={settingsRef} onClick={onSettingsClick} ></button>
+      <ContextMenu opened={isSettingsOpen && (undefined !== hasRangeData)} onOutsideClick={handleCloseSetting} >
+        <ContextMenuItem key={0} className={hasRangeData ? "" : "core-context-menu-disabled"} onSelect={handleZoomToLayer}>{labelZoomToLayer}</ContextMenuItem>
```

Comment:
Since no icons are used in this menu, the hideIconContainer prop should be set to avoid the empty area on the left of the dropdown. For example:
``ts
        <ContextMenuItem hideIconContainer key={0} className={hasRangeData ? "" : "core-context-menu-disabled"} onSelect={handleZoomToLayer}>{labelZoomToLayer}</ContextMenuItem>
        <ContextMenuItem hideIconContainer key={1} onSelect={handleRemoveLayer}>{labelDetach}</ContextMenuItem>
        <ContextMenuItem hideIconContainer key={2} >
          <Slider min={0} max={100} values={[transparency * 100]} step={1} showTooltip showMinMax onChange={handleTransparencyChange} />
        </ContextMenuItem>
```

Reply:
Done.